### ISSUE TITLE
Missing $INODE64 versions for readdir_r and fdopendir?

### ISSUE THREAD

**Issue by dkegel-fastly** [aykevl/macos-minimal-sdk#1]
Over in tinygo, https://github.com/tinygo-org/tinygo/pull/2644 shows two interesting errors:
```
ld.lld: error: undefined symbol: _readdir_r$INODE64
tinygo:ld.lld: error: undefined symbol: _fdopendir$INODE64
```
Those errors go away if you revert https://github.com/tinygo-org/tinygo/pull/2586

Those function calls have been in use since https://github.com/tinygo-org/tinygo/pull/2499 landed.

**Comment by aykevl** [comment]
Shorter reproducer: `GOOS=darwin tinygo test -c compress/flate`

**Comment by aykevl** [comment]
Fixed with ebb736fda2bec7cea38dcda807518b835a539525. It was basically just a matter of adding the dirent.h header file.

**Comment by aykevl** [comment]
In case such issues occur in the future, these are the steps I took:

- I looked up the header file for the missing symbols. For both readdir_r and fdopendir it is <dirent.h>.
- I added this header to src/libSystem.h.
- I ran `./update.sh` to update all generated files.
- I found that sys/dirent.h was missing. I found it in the XNU source code, so I added that header to update.sh
- I re-ran `./update.sh`. This time it succeeded.
- I checked the generated code (to check that everything is under an open source license) and committed the result.

In some cases these header files need a bit more massaging, see update.sh.
